### ISSUE TITLE
axe.utils.querSelectorAll() fails to ID query a slotted light dom node

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4346]
The following shouldn't return an empty array:

```html
<host-elm>
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <h2 id="target">Light content</h2>
</host-elm>
<script>
  const root = axe.setup()
  axe.utils.querySelectorAll('#target') // Should not be empty
</script>
```

I don't think this can impact how axe-core runs, as we never user ID queries in rules. But its a bit of an annoyance when writing tests.
